lib/dialects/oracledb/index.js

```diff
@@ -76,27 +76,30 @@ Client_Oracledb.prototype.prepBindings = function(bindings) {
 // connection needs to be added to the pool.
 Client_Oracledb.prototype.acquireRawConnection = function() {
   const client = this;
-  const asyncConnection = new Bluebird(function(resolver, rejecter) {
+  const asyncConnection = new Bluebird(async function(resolver, rejecter) {
     // If external authentication dont have to worry about username/password and
     // if not need to set the username and password
-    const oracleDbConfig = client.connectionSettings.externalAuth
-      ? { externalAuth: client.connectionSettings.externalAuth }
+    const connectionSettings = await client
```

Comment:
do not use async inside new promise callback
it will not catch errors, so you get unhandledRejection

Reply:
@Ali-Dalal This needs to be addressed